src/llm_utils.py

```diff
@@ -5,178 +5,136 @@
-    for block in comment_blocks:
-        block = block.strip()
-        if not block:
+    for block in response.split('---'):
+        lines = [line.strip() for line in block.split('\n') if line.strip()]
+        if not lines:
             continue
-            
-        comment_data = {
+
+        comment = {
             'path': None,
-            'line': None,
-            'body': ''
+            'start_line': None,
+            'end_line': None,
+            'body': []
         }
+        valid = True
         
-        # Validate required fields
-        has_file = False
-        has_line = False
-        has_comment = False
-        
-        for line in block.split('\n'):
-
-            line = line.strip()
-
+        # Parse block
+        for line in lines:
             if line.startswith('FILE:'):
-                comment_data['path'] = line.split('FILE:', 1)[-1].strip()
-                has_file = True
-
-            elif line.startswith('LINE:'):
+                comment['path'] = line.split('FILE:', 1)[-1].strip()
+            elif line.startswith('LINES:'):
                 try:
-                    comment_data['line'] = int(line.split('LINE:', 1)[-1].strip())
-                    has_line = True
+                    parts = line.split('LINES:', 1)[-1].strip().split('-')
+                    start = int(parts[0])
+                    end = int(parts[1])
+                    comment['start_line'], comment['end_line'] = sorted([start, end])
                 except (ValueError, IndexError):
-                    continue
-
+                    valid = False
             elif line.startswith('COMMENT:'):
-                comment_data['body'] += line.split('COMMENT:', 1)[-1].strip() + '\n'
-                has_comment = True
-
+                comment['body'].append(line.split('COMMENT:', 1)[-1].strip())
             elif line.startswith('SUGGESTION:'):
                 suggestion = line.split('SUGGESTION:', 1)[-1].strip()
                 if suggestion and suggestion != 'N/A':
-                    comment_data['body'] += f"\n```suggestion\n{suggestion}\n```"
+                    comment['body'].append(f'```suggestion\n{suggestion}\n```')
+            else:
+                comment['body'].append(line)
 
-            elif line:
-                comment_data['body'] += line + '\n'
-        
-        # Only add properly formatted comments
-        if has_file and has_line and has_comment:
-            comment_data['body'] = comment_data['body'].strip()
-            review_comments.append(comment_data)
+        # Validate required fields
+        if not all([
+            comment['path'],
+            isinstance(comment['start_line'], int),
+            isinstance(comment['end_line'], int),
+            comment['start_line'] <= comment['end_line']
+        ]):
+            valid = False
+
+        if valid:
+            review_comments.append({
+                'path': comment['path'],
+                'start_line': comment['start_line'],
```

Comment:
Adding comments in the code
```suggestion
|
```
N/A